src/ctia/entity/note.clj

```diff
@@ -0,0 +1,124 @@
+(ns ctia.entity.note
+  (:require [ctia.domain.entities :refer [page-with-long-id un-store-page]]
+            [ctia.entity.note.schemas :as note-schemas]
+            [ctia.http.routes.common :as routes.common]
+            [ctia.http.routes.crud :refer [services->entity-crud-routes]]
+            [ctia.lib.compojure.api.core :refer [GET routes]]
+            [ctia.schemas.core :refer [APIHandlerServices]]
+            [ctia.store :refer [list-records]]
+            [ctia.stores.es.mapping :as em]
+            [ctia.stores.es.store :refer [def-es-store]]
+            [schema-tools.core :as st]
+            [schema.core :as s]))
+
+(def note-mapping
+  {"note"
+   {:dynamic false
+    :properties
+    (merge
+     em/base-entity-mapping
+     em/sourcable-entity-mapping
+     em/stored-entity-mapping
+     {:entity_id em/token
+      :content em/searchable-token})}})
+
+(def-es-store NoteStore
+  :note
+  note-schemas/StoredNote
+  note-schemas/PartialStoredNote)
+
+(def note-sort-fields
+  (apply s/enum note-schemas/note-fields))
+
+(s/defschema NoteFieldsParam
+  {(s/optional-key :fields) [note-sort-fields]})
+
+(s/defschema NoteQueryParams
+  (st/merge
+   NoteFieldsParam
+   routes.common/PagingParams
+   {:entity_id s/Str
+    (s/optional-key :sort_by) note-sort-fields}))
+
+(def NoteGetParams NoteFieldsParam)
+
+(s/defschema NoteByExternalIdQueryParams
+  (st/dissoc NoteQueryParams :entity_id))
+
+(def capabilities
+  #{:create-note
+    :read-note
+    :delete-note
+    :search-note
+    :list-notes})
+
+(def searchable-fields
+  #{:id
+    :source
+    :entity_id
+    :content})
+
+(s/defn NoteSearchParams :- (s/protocol s/Schema)
+  [services :- APIHandlerServices]
+  (st/merge
+   routes.common/PagingParams
+   routes.common/BaseEntityFilterParams
+   routes.common/SourcableEntityFilterParams
+   routes.common/SearchableEntityParams
+   note-schemas/NoteFieldsParam
+   (st/optional-keys
+    {:entity_id s/Str
+     :content s/Str})))
```

Comment:
services is not used (and you can move it to the schemas ns)
```suggestion
(s/defschema NoteSearchParams
  (st/merge
   routes.common/PagingParams
   routes.common/BaseEntityFilterParams
   routes.common/SourcableEntityFilterParams
   routes.common/SearchableEntityParams
   note-schemas/NoteFieldsParam
   (st/optional-keys
    {:entity_id s/Str
     :content s/Str}))
```

Reply:
I refactored the ns and moved everything schema related to the relevant ns